### ISSUE TITLE
Error on Player if no audio output device is accessable

### ISSUE THREAD

**Issue by finn307** [pyglet/pyglet#231]
Running audio (tested with wav) via a player causes following error immediately:
```
Exception ignored in: <function DirectSoundDriver._del_ at 0x0D05B7C0>
Traceback (most recent call last):
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\media\drivers\directsound\interface.py", line 78, in _del_
    self._native_dsound.Release()
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\com.py", line 149, in _call
    ret = self.method.get_field()(self.i, self.name)(obj, *args)
ValueError: NULL COM pointer access
```
This leads to an Error later on (probably when source stops playing):
```
Traceback (most recent call last):
  File "C:/Users/Finn/Data/PythonProjects/PuzzleDefense/main.py", line 546, in <module>
    pyglet.app.run()
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\app\_init_.py", line 107, in run
    event_loop.run()
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\app\base.py", line 167, in run
    timeout = self.idle()
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\app\base.py", line 237, in idle
    redraw_all = self.clock.call_scheduled_functions(dt)
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\clock.py", line 292, in call_scheduled_functions
    item.func(now - item.last_ts, *item.args, **item.kwargs)
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\media\player.py", line 221, in <lambda>
    pyglet.clock.schedule_once(lambda dt: self.dispatch_event("on_eos"), source.duration)
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\event.py", line 420, in dispatch_event
    raise e
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\event.py", line 415, in dispatch_event
    if getattr(self, event_type)(*args):
  File "C:\Users\Finn\Data\PythonProjects\PuzzleDefense\venv\lib\site-packages\pyglet\media\player.py", line 609, in on_eos
    self._audio_player.clear()
AttributeError: 'NoneType' object has no attribute 'clear'
Error in sys.excepthook:

Original exception was:

```

It looks like an error occurse in `__del__` so defaulting to an silent audio driver is getting prevented.

pyglet version 1.5.5
Windows 10 Version 1909

**Comment by benmoran56** [comment]
This should now be fixed in pyglet v1.5.7, which was just release. Please let me know if you still see issues after upgrading.

**Comment by finn307** [comment]
It's all working fine. No problems.
